apps/dapp/src/components/Pages/Core/NewUI/DefendModal.tsx

```diff
@@ -0,0 +1,195 @@
+import { formatToken } from 'utils/formatter';
+import env from 'constants/env';
+import { TempleStableAMMRouter__factory } from 'types/typechain';
+import { TreasuryIV__factory } from 'types/typechain/typechain';
+import { useNotification } from 'providers/NotificationProvider';
+import { useDebouncedCallback } from 'use-debounce';
+
+interface IProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const { DAI, TEMPLE_TOKEN } = TICKER_SYMBOL;
+
+const inputConfig: CryptoValue = {
+  kind: 'value',
+  value: `${TEMPLE_TOKEN}`,
+};
+
+const outputConfig: CryptoSelector = {
+  kind: 'select',
+  cryptoOptions: [DAI],
+  onCryptoChange: () => {},
+  selected: DAI,
+};
+
+export const DefendModal: React.FC<IProps> = ({ isOpen, onClose }) => {
+  const { wallet, signer, balance } = useWallet();
+  const { openNotification } = useNotification();
+  const [inputValue, setInputValue] = useState('');
+  const [iv, setIv] = useState(ZERO);
+  const [quote, setQuote] = useState<BigNumber>();
+
+  // Set intrinsic value on mount
+  useEffect(() => {
+    const getIv = async () => {
+      if (!signer) return;
+      const treasuryIv = new TreasuryIV__factory(signer).attach(env.contracts.treasuryIv);
+      const { frax, temple } = await treasuryIv.intrinsicValueRatio();
+      setIv(frax.div(temple));
+    };
+    getIv();
+  }, [signer]);
+
+  // Fetch quote, debounced
+  const debouncedFetchQuote = useDebouncedCallback(async (amount: string) => {
+    const quote = await fetchQuote(inputValue);
+    setQuote(quote);
+  }, 750);
+
+  // Refresh quote on input change
+  useEffect(() => {
+    debouncedFetchQuote(inputValue);
+  }, [inputValue]);
+
+  // Get quote from router
+  const fetchQuote = async (input: string): Promise<BigNumber> => {
+    if (!signer || !wallet) return ZERO;
+    const router = new TempleStableAMMRouter__factory(signer).attach(env.contracts.templeV2Router);
+    const amountIn = getBigNumberFromString(input, 18);
+    if (amountIn.lte(ZERO)) return ZERO;
+    const { amountOut, priceBelowIV } = await router.swapExactTempleForStableQuote(
+      env.contracts.templeV2FraxPair,
+      amountIn
+    );
+    return priceBelowIV ? amountIn.mul(10_000).div(iv) : amountOut;
+  };
+
+  // Sell TEMPLE
+  const sell = async () => {
+    // Initialize router contract
+    if (!signer || !wallet) return;
+    const router = new TempleStableAMMRouter__factory(signer).attach(env.contracts.templeV2Router);
+
+    // Build swap parameters
+    const amountOut = await fetchQuote(inputValue);
+    if (!amountOut) return;
+    const fraxAddress = env.contracts.frax; // Use FRAX address even though Defend returns DAI
+    const deadline = 20;
+    const deadlineTimestamp = Math.floor(Date.now() / 1000) + deadline * 60;
+    const amountIn = getBigNumberFromString(inputValue, 18);
+    const slippage = 0.5;
+    const slippageBps = BigNumber.from(10_000 + 100 * slippage);
+    const minAmountOut = amountOut.mul(10_000).div(slippageBps);
```

Comment:
There's no slippage in Defend. This could be:

```
const minAmountOut = priceBelowIV ? amountOut : amountOut.mul(10_000).div(slippageBps);
```

Reply:
keeping slippage calc to keep the logic simple